crates/dbs-boot/src/aarch64/mod.rs

```diff
@@ -27,3 +36,37 @@ impl From<VmFdtError> for Error {
         Error::CreateFdt(e)
     }
 }
+
+/// Returns the memory address where the kernel could be loaded.
+pub fn get_kernel_start() -> u64 {
+    layout::DRAM_MEM_START
+}
+
+// Auxiliary function to get the address where the device tree blob is loaded.
+fn get_fdt_addr<M: GuestMemory>(mem: &M) -> u64 {
+    // If the memory allocated is smaller than the size allocated for the FDT,
+    // we return the start of the DRAM so that
+    // we allow the code to try and load the FDT.
+    if let Some(offset) = mem.last_addr().checked_sub(layout::FDT_MAX_SIZE as u64 - 1) {
+        if mem.address_in_range(offset) {
+            return offset.raw_value();
+        }
+    }
+    crate::layout::DRAM_MEM_START
+}
+
+/// Returns the memory address where the initrd could be loaded.
+pub fn initrd_load_addr<M: GuestMemory>(guest_mem: &M, initrd_size: u64) -> super::Result<u64> {
+    let round_to_pagesize =
+        |size| (size + (PAGE_SIZE as u64 - 1)) & !(PAGE_SIZE as u64 - 1);
+    match GuestAddress(get_fdt_addr(guest_mem)).checked_sub(round_to_pagesize(initrd_size) as u64) {
+        Some(offset) => {
+            if guest_mem.address_in_range(offset) {
+                return Ok(offset.raw_value());
+            } else {
+                return Err(Error::InitrdAddress);
+            }
+        }
+        None => return Err(Error::InitrdAddress),
```

Comment:
Is the return here needed?

Reply:
If the return is added, cacrgo build will report unreachable return. So there is no need return.

Reply:
done, I have delete the return key word.